### PR TITLE
CPK support for datalake

### PR DESCRIPTION
**Pull request by ashruti-msft**
Added cpk support for datalake

### CODE REVIEW

File: component/azstorage/datalake.go [hunk 1]
```diff
@@ -58,10 +58,11 @@ import (
...
 	AzStorageConnection
-	Auth       azAuth
-	Service    *service.Client
-	Filesystem *filesystem.Client
-	BlockBlob  BlockBlob
+	Auth           azAuth
+	Service        *service.Client
+	Filesystem     *filesystem.Client
+	BlockBlob      BlockBlob
+	datalakeCPKOpt *directory.CPKInfo
```
Comment (souravgupta-msft) [inline component/azstorage/datalake.go:65]: Please add changelog and tests for the datalake APIs where CPK is added.

File: component/azstorage/datalake.go [hunk 2]
```diff
@@ -348,7 +361,9 @@ func (dl *Datalake) GetAttr(name string) (attr *internal.ObjAttr, err error) {
 	log.Trace("Datalake::GetAttr : name %s", name)
 
 	fileClient := dl.Filesystem.NewFileClient(filepath.Join(dl.Config.prefixPath, name))
-	prop, err := fileClient.GetProperties(context.Background(), nil)
+	prop, err := fileClient.GetProperties(context.Background(), &file.GetPropertiesOptions{
```
Comment (vibhansa-msft) [inline component/azstorage/datalake.go:364]: Other than rename and getProperties we do not need the CPK in any other call. What about Set/Get AccessControl apis.

Comment (ashruti-msft) [reply component/azstorage/datalake.go:364]: There is no field for CPK in the options for those apis

File: component/azstorage/datalake.go (hunk 1 above)
Comment (ashruti-msft) [reply component/azstorage/datalake.go]: I have added them pls review

**Review by vibhansa-msft — APPROVED** [review]

**Review by souravgupta-msft — APPROVED** [review]

**Pull request merged**